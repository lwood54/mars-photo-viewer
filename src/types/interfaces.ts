export interface Camera {
  full_name: string;
  id: number;
  name: string;
  rover_id: number;
}

export interface Rover {
  id: number;
  landing_date: string;
  launch_date: string;
  name: string;
  status: string;
}

export interface PhotoData {
  camera?: Camera;
  earth_date?: string;
  id: number;
  img_src: string;
  rover: Rover;
  sol?: number;
}

export interface PhotoViewerState {
  photoData: PhotoData | null;
  selCamera: string;
  selSol: number;
  selEarthDate: string;
  earthYear: string;
  earthMonth: string;
  earthDay: string;
  selRoverType: string;
}

export interface Action {
	type: string;
	payload: any;
}
