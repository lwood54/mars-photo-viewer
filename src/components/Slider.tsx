import React, { useState, useEffect } from "react";
import styled, { keyframes } from "styled-components";

// create keyframes for animation injection
const anim_center_left = keyframes`
  from {
    transform: translate(0, 0);
  }
  to {
    transform: translate(-71vw, 0);
  }
`;
// apply animations, this will be used when items move to the left
const Center_left = styled.div`
  animation: ${anim_center_left} 500ms ease-in;
  animation-fill-mode: forwards;
`;

const anim_center_right = keyframes`
	from {
    transform: translate(0, 0);
  }
  to {
    transform: translate(71vw, 0);
  }
`;
const Center_right = styled.div`
  animation: ${anim_center_right} 500ms ease-in;
  animation-fill-mode: forwards;
`;

// container for slider images
const SliderSC = styled.div`
  display: flex;
  justify-content: center;
  overflow: hidden;
  max-height: 500px;
`;

// this will be applied if there is not active animation happening
// need to reset class so animations will occur on next click
const Center_normal = styled.div`
  animation: none;
`;

const ArrowContainerSC = styled.div`
  width: 80%;
  margin: 1rem auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

// setting base arrow characteristics that can be manipulated for arrows
const ArrowBaseSC = styled.span`
  display: inline-block;
  width: 15px;
  height: 15px;
  border-top: 2px solid #000;
  border-right: 2px solid #000;
  cursor: pointer;
`;

const PrevArrowSC = styled(ArrowBaseSC)`
  transform: rotate(-135deg);
`;

const NextArrowSC = styled(ArrowBaseSC)`
  transform: rotate(45deg);
`;

const NoSliderSC = styled.div`
  /* width: 100px;
	margin: auto; */
`;

const BasePhotoCont = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  img {
    width: 100%;
  }
`;

const InfoContainer = styled.div`
  width: 100%;
  margin: 0.75rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
`;

const PhotoDetail = styled.div`
  width: 50%;
  margin: 0.5rem;
  text-align: center;
  min-width: 500px;
`;

// container of the display photo
// padding is required in order to push previous and next divs out of viewport
const PhotoContainerSC = styled(BasePhotoCont)`
  padding: 0 25%;
`;

// Slider will take an array of Photos
function Slider({ photoArray }: { photoArray: JSX.Element[] }): JSX.Element {
  // copying photo array so it can be mutated and displayed as user clicks prev or next
  const [baseArray, setBaseArray] = useState<Array<JSX.Element>>(photoArray);
  // tracking whether elements are in the act of moving, which will disable click functions
  // as well as determine containers/classes to be used for animation purposes
  const [moveLeft, setMoveLeft] = useState<boolean>(false);
  const [moveRight, setMoveRight] = useState<boolean>(false);
  // tracking which image user is on because array is mutated
  const [photoCount, setPhotoCount] = useState(1);

  useEffect(() => {
    // process array to show only have first, next, and last items
    if (photoArray) {
      setPhotoCount(1);
      setBaseArray(photoArray);
    }
  }, [photoArray]);

  // const clickRight = (prevArray: JSX.Element[] | null): JSX.Element[] | null => {
  const clickRight = () => {
    // take array param, move all items to left, 0 goes to end
    let array: JSX.Element[] | null = null;
    if (baseArray) {
      array = [...baseArray];
      let firstItem = array.splice(0, 1);
      array.push(firstItem[0]);
    }
    // ensuring flag is reset
    setMoveRight(false);
    // flagging movement
    setMoveLeft(true);
    // need to wait for animation to progress, then setting the shuffled array
    // which will display photos in their new locations, which are the same spots
    // that the animations end
    setTimeout(() => {
      setMoveLeft(false);
      setBaseArray(array);
    }, 500);
    if (photoCount === baseArray.length) {
      setPhotoCount(1);
    } else {
      setPhotoCount(photoCount + 1);
    }
  };
  // const clickLeft = (prevArray: JSX.Element[] | null): JSX.Element[] | null => {
  const clickLeft = () => {
    let array: JSX.Element[] | null = null;
    if (baseArray) {
      array = [...baseArray];
      let lastItem = array.pop();
      array.unshift(lastItem!);
    }
    // similar functionality as clickRight, but array mutation was handles slightly differently
    setMoveLeft(false);
    setMoveRight(true);
    setTimeout(() => {
      setMoveRight(false);
      setBaseArray(array);
    }, 500);
    if (photoCount === 1) {
      setPhotoCount(baseArray.length);
    } else {
      setPhotoCount(photoCount - 1);
    }
  };
  // different styled components are displayed based on required animation and current
  // motion status
  return (
    <div>
      <ArrowContainerSC>
        <PrevArrowSC onClick={moveLeft || moveRight ? () => false : clickLeft}></PrevArrowSC>
        <NextArrowSC onClick={moveLeft || moveRight ? () => false : clickRight}></NextArrowSC>
      </ArrowContainerSC>
      {baseArray?.length > 2 ? (
        <SliderSC>
          <BasePhotoCont>
            {moveLeft ? (
              <Center_left>{baseArray ? baseArray[baseArray.length - 1] : undefined}</Center_left>
            ) : moveRight ? (
              <Center_right>{baseArray ? baseArray[baseArray.length - 1] : undefined}</Center_right>
            ) : (
              <Center_normal>{baseArray ? baseArray[baseArray.length - 1] : undefined}</Center_normal>
            )}
          </BasePhotoCont>
          <PhotoContainerSC>
            {moveLeft ? (
              <Center_left>{baseArray ? baseArray[0] : undefined}</Center_left>
            ) : moveRight ? (
              <Center_right>{baseArray ? baseArray[0] : undefined}</Center_right>
            ) : (
              <Center_normal>{baseArray ? baseArray[0] : undefined}</Center_normal>
            )}
          </PhotoContainerSC>
          <BasePhotoCont>
            {moveLeft ? (
              <Center_left>{baseArray ? baseArray[1] : undefined}</Center_left>
            ) : moveRight ? (
              <Center_right>{baseArray ? baseArray[1] : undefined}</Center_right>
            ) : (
              <Center_normal>{baseArray ? baseArray[1] : undefined}</Center_normal>
            )}
          </BasePhotoCont>
        </SliderSC>
      ) : (
        <SliderSC>
          <div>{baseArray ? baseArray[0] : null}</div>
          <div>{baseArray ? baseArray[1] : null}</div>
        </SliderSC>
      )}

      <InfoContainer>
        <PhotoDetail>
          {photoCount} / {baseArray ? baseArray.length : 0}
        </PhotoDetail>
        <PhotoDetail>
          <strong>Rover:</strong> {baseArray ? baseArray[0]?.props.photoData.rover.name : ""}
        </PhotoDetail>
        <PhotoDetail>
          <strong>Camera: </strong>
          {baseArray ? baseArray[0]?.props.photoData.camera.full_name : ""}
        </PhotoDetail>
        <PhotoDetail>
          <strong>Earth Date: </strong>
          {baseArray ? baseArray[0]?.props.photoData.earth_date : ""}
        </PhotoDetail>
      </InfoContainer>
    </div>
  );
}

export default Slider;
