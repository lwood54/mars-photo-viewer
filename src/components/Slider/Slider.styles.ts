import styled, {keyframes} from "styled-components";

// create keyframes for animation injection
export const anim_center_left = keyframes`
  from {
    transform: translate(0, 0);
  }
  to {
    transform: translate(-71vw, 0);
  }
`;
// apply animations, this will be used when items move to the left
export const CenterLeft = styled.div`
  animation: ${anim_center_left} 500ms ease-in;
  animation-fill-mode: forwards;
  @media (max-width: 750px) {
    animation: ${anim_center_left} 300ms ease-in;
  }
`;

export const anim_center_right = keyframes`
	from {
    transform: translate(0, 0);
  }
  to {
    transform: translate(71vw, 0);
  }
`;
export const CenterRight = styled.div`
  animation: ${anim_center_right} 500ms ease-in;
  animation-fill-mode: forwards;
  @media (max-width: 750px) {
    animation: ${anim_center_right} 300ms ease-in;
  }
`;

// container for slider images
export const SliderSC = styled.div`
  display: flex;
  justify-content: center;
  margin-top: -20px;
  img {
    cursor: auto;
  }
`;

// this will be applied if there is not active animation happening
// need to reset class so animations will occur on next click
export const CenterNormal = styled.div`
  animation: none;
`;

export const ArrowContainerSC = styled.div`
  width: 80%;
  margin: 0 auto 0.5rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

// setting base arrow characteristics that can be manipulated for arrows
export const ArrowBaseSC = styled.span`
  display: inline-block;
  width: 15px;
  height: 15px;
  border-top: 2px solid #000;
  border-right: 2px solid #000;
  cursor: pointer;
  &:hover {
    border-top: 3px solid #000;
    border-right: 3px solid #000;
  }
  &:active {
    border-top: 2px solid #000;
    border-right: 2px solid #000;
  }
`;

export const PrevArrowSC = styled(ArrowBaseSC)`
  transform: rotate(-135deg);
`;

export const NextArrowSC = styled(ArrowBaseSC)`
  transform: rotate(45deg);
`;

export const BasePhotoCont = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  img {
    width: auto;
  }
`;

export const InfoContainer = styled.div`
  margin: 0.5rem;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  box-sizing: border-box;
`;

export const PhotoDetail = styled.div<{ photoCount?: boolean }>`
  width: 50%;
  margin: 0.5rem;
  margin-bottom: 0;
  margin: ${(props) => (props.photoCount ? "auto" : "0 0.5rem 0.25rem")}; // use auto margin for photo count
  text-align: center;
  min-width: 500px;
`;

// container of the display photo
// padding is required in order to push previous and next divs out of viewport
export const PhotoContainerSC = styled(BasePhotoCont)`
  padding: 0 25%;
  @media (max-width: 800px) {
    padding: 0 55%;
  }
`;