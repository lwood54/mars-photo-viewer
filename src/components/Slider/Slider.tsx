import React, { useState, useEffect } from "react";
import {
  PhotoDetail,
  ArrowContainerSC,
  PrevArrowSC,
  NextArrowSC,
  SliderSC,
  BasePhotoCont,
  CenterLeft,
  CenterRight,
  CenterNormal,
  PhotoContainerSC,
  InfoContainer,
} from "./Slider.styles";

// Slider will take an array of Photos
function Slider({ photoArray }: { photoArray: JSX.Element[] }): JSX.Element {
  // copying photo array so it can be mutated and displayed as user clicks prev or next
  const [basePhotoArray, setBasePhotoArray] = useState<Array<JSX.Element>>(photoArray);
  // tracking whether elements are in the act of moving, which will disable click functions
  // as well as determine containers/classes to be used for animation purposes
  const [moveLeft, setMoveLeft] = useState<boolean>(false);
  const [moveRight, setMoveRight] = useState<boolean>(false);
  // tracking which image user is on because array is mutated
  const [photoCount, setPhotoCount] = useState(1);

  useEffect(() => {
    // process array to show only have first, next, and last items
    if (photoArray) {
      setPhotoCount(1);
      setBasePhotoArray(photoArray);
    }
  }, [photoArray]);

  // const clickRight = (prevArray: JSX.Element[] | null): JSX.Element[] | null => {
  const clickRight = () => {
    // take array param, move all items to left, 0 goes to end
    let array: JSX.Element[] | null = null;
    if (basePhotoArray) {
      array = [...basePhotoArray];
      let firstItem = array.splice(0, 1);
      array.push(firstItem[0]);
    }
    // ensuring flag is reset
    setMoveRight(false);
    // flagging movement
    setMoveLeft(true);
    // need to wait for animation to progress, then setting the shuffled array
    // which will display photos in their new locations, which are the same spots
    // that the animations end
    let timeOut = window.innerWidth <= 750 ? 300 : 500;
    setTimeout(() => {
      setMoveLeft(false);
      setBasePhotoArray(array ?? []);
    }, timeOut);
    if (photoCount === basePhotoArray.length) {
      setPhotoCount(1);
    } else {
      setPhotoCount(photoCount + 1);
    }
  };
  // const clickLeft = (prevArray: JSX.Element[] | null): JSX.Element[] | null => {
  const clickLeft = () => {
    let array: JSX.Element[] | null = null;
    if (basePhotoArray) {
      array = [...basePhotoArray];
      let lastItem = array.pop();
      array.unshift(lastItem!);
    }
    // similar functionality as clickRight, but array mutation was handles slightly differently
    setMoveLeft(false);
    setMoveRight(true);
    let timeOut = window.innerWidth <= 750 ? 300 : 500;
    setTimeout(() => {
      setMoveRight(false);
      setBasePhotoArray(array ?? []);
    }, timeOut);
    if (photoCount === 1) {
      setPhotoCount(basePhotoArray.length);
    } else {
      setPhotoCount(photoCount - 1);
    }
  };
  // different styled components are displayed based on required animation and current
  // motion status
  return (
    <>
      <PhotoDetail photoCount>
        {/* {photoCount} / {basePhotoArray ? basePhotoArray.length : 0} */}
        {photoCount} / {basePhotoArray && basePhotoArray.length}
      </PhotoDetail>
      <ArrowContainerSC>
        <PrevArrowSC onClick={moveLeft || moveRight ? () => false : clickLeft}></PrevArrowSC>
        <NextArrowSC onClick={moveLeft || moveRight ? () => false : clickRight}></NextArrowSC>
      </ArrowContainerSC>
      {basePhotoArray?.length > 2 ? (
        <SliderSC>
          <BasePhotoCont>
            {moveLeft ? (
              <CenterLeft>{basePhotoArray && basePhotoArray[basePhotoArray.length - 1]}</CenterLeft>
            ) : moveRight ? (
              <CenterRight>{basePhotoArray && basePhotoArray[basePhotoArray.length - 1]}</CenterRight>
            ) : (
              <CenterNormal>{basePhotoArray && basePhotoArray[basePhotoArray.length - 1]}</CenterNormal>
            )}
          </BasePhotoCont>
          <PhotoContainerSC data-testid="main-view">
            {moveLeft ? (
              <CenterLeft>{basePhotoArray && basePhotoArray[0]}</CenterLeft>
            ) : moveRight ? (
              <CenterRight>{basePhotoArray && basePhotoArray[0]}</CenterRight>
            ) : (
              <CenterNormal>{basePhotoArray && basePhotoArray[0]}</CenterNormal>
            )}
          </PhotoContainerSC>
          <BasePhotoCont>
            {moveLeft ? (
              <CenterLeft>{basePhotoArray && basePhotoArray[1]}</CenterLeft>
            ) : moveRight ? (
              <CenterRight>{basePhotoArray && basePhotoArray[1]}</CenterRight>
            ) : (
              <CenterNormal>{basePhotoArray && basePhotoArray[1]}</CenterNormal>
            )}
          </BasePhotoCont>
        </SliderSC>
      ) : (
        <SliderSC>
          <div>{basePhotoArray && basePhotoArray[0]}</div>
          <div>{basePhotoArray && basePhotoArray[1]}</div>
        </SliderSC>
      )}

      <InfoContainer>
        <PhotoDetail>
          <strong>Rover:</strong> {basePhotoArray && basePhotoArray[0]?.props.photoData.rover.name}
        </PhotoDetail>
        <PhotoDetail>
          <strong>Camera: </strong>
          {basePhotoArray && basePhotoArray[0]?.props.photoData.camera.full_name}
        </PhotoDetail>
        <PhotoDetail>
          <strong>Earth Date: </strong>
          {basePhotoArray && basePhotoArray[0]?.props.photoData.earth_date}
        </PhotoDetail>
      </InfoContainer>
    </>
  );
}

export default Slider;
