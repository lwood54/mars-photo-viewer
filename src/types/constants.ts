export enum camera {
    ALL = "all",
    FHAZ = "fhaz",
    RHAZ = "rhaz",
    MAST = "mast",
    CHEMCAM = "chemcam",
    MAHLI = "mahli",
    MARDI = "mardi",
    NAVCAM = "navcam",
    PANCAM = "pancam",
    MINITES = "minites",
  }

	export enum viewOptions {
		GRID = "grid",
		SLIDER = "slider"
	}

	export enum ACTIONS {
		SET_PHOTO_DATA = "SET_PHOTO_DATA",
		SET_CAMERA = "SET_CAMERA",
		SET_SOL = "SET_SOL",
		SET_EARTH_DATE = "SET_EARTH_DATE",
		SET_EARTH_YEAR = "SET_EARTH_YEAR",
		SET_EARTH_MONTH = "SET_EARTH_MONTH",
		SET_EARTH_DAY = "SET_EARTH_DAY",
		SET_ROVER_TYPE = "SET_ROVER_TYPE"
	}