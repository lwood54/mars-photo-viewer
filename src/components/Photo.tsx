import React from "react";
import { PhotoData } from "../types/interfaces";
import styled from "styled-components";

// TS issues with styled-components when passing props unless
// SC is defined by type
type PhotItemDiv = {
  currentView: "grid" | "slider";
};

// styled-component decorating phot item
const PhotoItem = styled.div<PhotItemDiv>`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 5px;
  width: ${(props) => (props.currentView === "grid" ? "250px" : "50vw")};
  margin: 5px;
  flex-wrap: wrap;
  box-sizing: border-box;
  cursor: pointer;
  img {
    max-width: 65%;
    height: auto;
  }
`;

// defining props that will be required to be passed
// data is an optional param on toggleModal
interface Props {
  photoData: PhotoData;
  key: number;
  toggleModal?: (e: React.MouseEvent<HTMLElement>, data?: PhotoData) => void;
  currentView: "grid" | "slider";
  // currentView: React.ElementType | keyof JSX.IntrinsicElements;
}

//
function Photo({ photoData, toggleModal, currentView }: Props): JSX.Element {
  // take in photo data, display image and content:
  // data to display:

  // passing data back onClick so that PhotoViewer can send the photoData
  // of the selected image to Modal, so Modal can display
  return (
    <div onClick={toggleModal ? (e) => toggleModal(e, photoData) : () => false}>
      <PhotoItem currentView={currentView}>
        <img src={photoData.img_src} alt={`${photoData?.rover.name} ${photoData?.earth_date}`} loading="lazy" />
      </PhotoItem>
    </div>
  );
}

export default Photo;
