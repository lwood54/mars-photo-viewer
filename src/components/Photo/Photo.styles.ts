import styled from "styled-components";
import {viewOptions} from "../../types/constants";

// TS issues with styled-components when passing props unless
// SC is defined by type
type PhotoItemDiv = {
  currentView: viewOptions.GRID | viewOptions.SLIDER;
};

// styled-component decorating phot item
export const PhotoItem = styled.div<PhotoItemDiv>`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 5px;
  width: ${(props) => (props.currentView === viewOptions.GRID ? "250px" : "50vw")};
  min-height: ${(props) => (props.currentView === viewOptions.GRID ? "auto" : "500px")};
  margin: 5px;
  flex-wrap: wrap;
  box-sizing: border-box;
  cursor: pointer;
  img {
    max-width: ${(props) => (props.currentView === viewOptions.GRID ? "100%" : "65%")};
    height: auto;
    @media (min-width: 1775px) {
      max-width: ${(props) => (props.currentView === viewOptions.GRID ? "100%" : "550px")};
    }
    @media (max-width: 1300px) {
      max-width: ${(props) => (props.currentView === viewOptions.GRID ? "auto" : "400px")};
    }
    @media (max-width: 775px) {
      /* max-width: ${(props) => (props.currentView === viewOptions.GRID ? "auto" : "350px")}; */
      width: 100%;
    }
  }
`;