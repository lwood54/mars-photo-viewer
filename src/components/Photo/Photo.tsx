import React from "react";
import { PhotoData } from "../../types/interfaces";
import { viewOptions } from "../../types/constants";
import { PhotoItem } from "./Photo.styles";

// defining props that will be required to be passed
// data is an optional param on toggleModal
interface Props {
  photoData: PhotoData;
  key: number;
  toggleModal?: (e: React.MouseEvent<HTMLElement>, data?: PhotoData) => void;
  currentView: viewOptions.GRID | viewOptions.SLIDER;
  // I thought this is most specific type which is even safer than string
  // so this would technically be of type string grid or type string slider
}

//
function Photo({ photoData, toggleModal, currentView }: Props): JSX.Element {
  // take in photo data, display image and content:
  // data to display:

  // passing data back onClick so that PhotoViewer can send the photoData
  // of the selected image to Modal, so Modal can display
  return (
    <div onClick={toggleModal ? (e) => toggleModal(e, photoData) : () => false}>
      <PhotoItem currentView={currentView}>
        <img src={photoData.img_src} alt={`${photoData?.rover.name} ${photoData?.earth_date}`} loading="lazy" />
      </PhotoItem>
    </div>
  );
}

export default Photo;
