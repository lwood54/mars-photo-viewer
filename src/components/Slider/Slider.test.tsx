import React from "react";
import { fireEvent, render, cleanup } from "@testing-library/react";
import Slider from "./Slider";
import Photo from "../Photo/Photo";
import { viewOptions } from "../../types/constants";

const mockPhoto1 = {
  camera: {
    full_name: "fhaz",
    id: 55,
    name: "front hazard camera",
    rover_id: 3,
  },
  earth_date: "string",
  id: 123,
  img_src: "first_photo_src",
  rover: {
    id: 22,
    landing_date: "2004-1-1",
    launch_date: "2003-10-5",
    name: "curiosity",
    status: "active",
  },
  sol: 34,
};

const mockPhoto2 = {
  amera: {
    full_name: "fhaz",
    id: 77,
    name: "front hazard camera",
    rover_id: 3,
  },
  earth_date: "2012-3-11",
  id: 345,
  img_src: "second_photo_src",
  rover: {
    id: 33,
    landing_date: "2005-2-2",
    launch_date: "2004-11-15",
    name: "curiosity",
    status: "active",
  },
  sol: 54,
};

const mockPhoto3 = {
  amera: {
    full_name: "front hazard camera",
    id: 88,
    name: "fhaz",
    rover_id: 2,
  },
  earth_date: "2013-4-12",
  id: 456,
  img_src: "third_photo_src",
  rover: {
    id: 44,
    landing_date: "2006-3-3",
    launch_date: "2005-12-1",
    name: "opportunity",
    status: "active",
  },
  sol: 876,
};

const mockPhotoArray = [
  <Photo photoData={mockPhoto1} key={1} currentView={viewOptions.SLIDER} />,
  <Photo photoData={mockPhoto2} key={2} currentView={viewOptions.SLIDER} />,
  <Photo photoData={mockPhoto3} key={3} currentView={viewOptions.SLIDER} />,
];

afterEach(cleanup);
// assert that given some data, slider will render that info
test("given data render slider pic with data", () => {
  const { getByTestId, debug, container } = render(<Slider photoArray={mockPhotoArray} />);
  let main_view = getByTestId("main-view");
  expect(container.firstChild).toMatchSnapshot();
  expect(main_view.innerHTML.indexOf('src="first_photo_src"')).toBeGreaterThan(-1);
});
