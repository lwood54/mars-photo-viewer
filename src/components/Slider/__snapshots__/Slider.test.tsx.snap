// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given data render slider pic with data 1`] = `
<div
  class="sc-jrsJCI jcQVQn"
>
  1
   / 
  3
</div>
`;
